spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/SSLConnectionTests.java

```diff
@@ -82,6 +89,69 @@ public void testNoAlgNoProps() throws Exception {
 	}
 
 	@Test
+	public void testKSTS() throws Exception {
+		RabbitConnectionFactoryBean fb = new RabbitConnectionFactoryBean();
+		Log logger = spy(TestUtils.getPropertyValue(fb, "logger", Log.class));
+		given(logger.isDebugEnabled()).willReturn(true);
+		new DirectFieldAccessor(fb).setPropertyValue("logger", logger);
+		ConnectionFactory rabbitCf = spy(TestUtils.getPropertyValue(fb, "connectionFactory", ConnectionFactory.class));
+		new DirectFieldAccessor(fb).setPropertyValue("connectionFactory", rabbitCf);
+		fb.setUseSSL(true);
+		fb.setKeyStoreType("JKS");
+		fb.setKeyStoreResource(new ClassPathResource("test.ks"));
+		fb.setKeyStorePassphrase("secret");
+		fb.setTrustStoreResource(new ClassPathResource("test.truststore.ks"));
+		fb.setKeyStorePassphrase("secret");
+		fb.setSecureRandom(SecureRandom.getInstanceStrong());
+		fb.afterPropertiesSet();
+		fb.getObject();
+		ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+		verify(logger).debug(captor.capture());
+		final String log = captor.getValue();
+		assertThat(log, allOf(containsString("KM: ["), containsString("TM: ["), containsString("random: java.")));
+	}
+
+	@Test
+	public void testNullTS() throws Exception {
+		RabbitConnectionFactoryBean fb = new RabbitConnectionFactoryBean();
+		Log logger = spy(TestUtils.getPropertyValue(fb, "logger", Log.class));
+		given(logger.isDebugEnabled()).willReturn(true);
+		new DirectFieldAccessor(fb).setPropertyValue("logger", logger);
+		ConnectionFactory rabbitCf = spy(TestUtils.getPropertyValue(fb, "connectionFactory", ConnectionFactory.class));
+		new DirectFieldAccessor(fb).setPropertyValue("connectionFactory", rabbitCf);
+		fb.setUseSSL(true);
+		fb.setKeyStoreType("JKS");
+		fb.setKeyStoreResource(new ClassPathResource("test.ks"));
+		fb.setKeyStorePassphrase("secret");
+		fb.afterPropertiesSet();
+		fb.getObject();
+		ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+		verify(logger).debug(captor.capture());
+		final String log = captor.getValue();
+		assertThat(log, allOf(containsString("KM: ["), containsString("TM: null"), containsString("random: null")));
+	}
+
+	@Test
+	public void testNullKS() throws Exception {
+		RabbitConnectionFactoryBean fb = new RabbitConnectionFactoryBean();
+		Log logger = spy(TestUtils.getPropertyValue(fb, "logger", Log.class));
+		given(logger.isDebugEnabled()).willReturn(true);
+		new DirectFieldAccessor(fb).setPropertyValue("logger", logger);
+		ConnectionFactory rabbitCf = spy(TestUtils.getPropertyValue(fb, "connectionFactory", ConnectionFactory.class));
+		new DirectFieldAccessor(fb).setPropertyValue("connectionFactory", rabbitCf);
+		fb.setUseSSL(true);
+		fb.setKeyStoreType("JKS");
```

Comment:
I see `KEY_STORE_DEFAULT_TYPE = "PKCS12"` is still not `JKS` for consistency and convenience.
I can't find external discussion, but seems for me we should fix that in `2.0`.
Right, not in this PR.